--- a/drivers//net/dsa/realtek/rtl8365mb.c	2025-04-07 16:06:37.000000000 +0800
+++ b/drivers//net/dsa/realtek/rtl8365mb.c	2025-04-27 13:54:22.346129104 +0800
@@ -1050,6 +1050,8 @@ static void rtl8365mb_phylink_get_caps(s
 	 * not necessarily the modes supported by the hardware, some of
 	 * which remain unimplemented.
 	 */
+        if (extint->supported_interfaces & RTL8365MB_PHY_INTERFACE_MODE_SGMII)
+                __set_bit(PHY_INTERFACE_MODE_SGMII, config->supported_interfaces);
 
 	if (extint->supported_interfaces & RTL8365MB_PHY_INTERFACE_MODE_RGMII)
 		phy_interface_set_rgmii(config->supported_interfaces);
